### PR TITLE
nodenext compatibility, correct typings

### PR DESCRIPTION
**Pull request by Uzlopak**
Part of [Migration of the infamous triplet to next level, Nodenext support ](https://github.com/fastify/fastify/issues/4349)

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
CI seems failing

**Comment by Uzlopak** [comment]
Ok, fixed the typings. They were from the beginning wrong. So I assume, nobody uses fast-json-stringify-compiler in a typescript environment. 

I had to change the purpose of some typings, which means, that it could be a breaking change. But on the other hand, as mentioned before, nobody complained about the fact that the typings were wrong, so probably nobody will complain anyway.

File: types/index.test-d.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      expectType<RouteDefinition>(routeOpts)
+      expectType<string>(schemaSerializerCode)
+    }
+  })
+  expectType<SerializerFactory>(factory)
+
+  const compiler = factory(schemaMap)
+  expectType<SerializerCompiler>(compiler)
+  expectType<Serializer>(compiler(endpointSchema))
+}
```
Comment (Uzlopak) [inline types/index.test-d.ts:139]:
```suggestion
}

```

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
Let's wait for @Eomm before landing this one.

**Review by Eomm — APPROVED** [review]

**Pull request merged**